docs/src/main/asciidoc/security-authentication-mechanisms.adoc

```diff
@@ -511,6 +511,88 @@ quarkus.http.auth.permission.bearer.auth-mechanism=bearer
 
 Ensure that the value of the `auth-mechanism` property matches the authentication scheme supported by `HttpAuthenticationMechanism`, for example, `basic`, `bearer`, or `form`.
 
+=== Endpoint-specific authentication mechanisms
```

Comment:
I am fine with propose title and I'll change it, but let me provide some context.

> This may read strange but the endpoint specific, path based authentication is already supported, it is really about using the annotations vs HTTP security policy to enable this feature.

Yes, but the annotation-based approach is more powerful because it is capable to select endpoint, which the HTTP Security Policy never will. You can have several endpoints with a same path and HTTP method but they will differ by content type etc. just FYI

Reply:
@michalvavrik I'm not disagreeing, I'm just saying that we already allow to do it with HTTP Security Policy, so when users will see one section "Path based authentication" where we say how to use HTTP Security Policy and `Endpoint specific authentication` where the annotations are used, they will be confused IMHO. So the suggestion is to make it clear that one section is about using the policy, another - annotations, to achieve the same goal

Reply:
@michalvavrik Ma be we can have a section `Path-based authentication` then the annotation based selection for JAX-RS goes next and we say we recommend it, the existing one for HTTP security policy goes last ? May be also clarify it as `Use annotations to enable path-based authentication for JAX-RS endpoints`